src/web/src/app/space/service.ts

```diff
@@ -363,73 +190,35 @@ export class SpaceService {
 		}
 	}
 
-	join(spaceId: string): Observable<boolean> {
-		let space = this.spaces.find(s => s.id === spaceId);
-		if (space) {
-			return of(space.requestsAllowed)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		} else {
-			return of(false)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		}
-	}
-
-	joinByLink(token: string): Observable<boolean> {
-		let link = this.links.find(l => l.id === token);
-		let now = new Date();
-		if (link.expiredAt > now) {
-			let space = this.spaces.find(s => s.id === link.spaceId);
-			return of(true)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		}
+	join(spaceId: string, userId: string): Observable<any> {
+		return this.http.post(this._apiUrl + "/invitation", {
+			id: 0,
+			spaceId: spaceId,
+			creatorId: userId,
+			userId: userId,
+			role: "guest",
+			createdAt: 0,
+		}, { observe: 'response' }).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		);
 	}
 
-	changeRoleInInvitation(id, newRole): Observable<boolean> {
-		let index = this.invitations.findIndex(i => i.id === id);
-		if (index > -1) {
-			this.invitations[index].role = newRole;
-			return of(true)
+	joinByLink(token: string): Observable<any> {
+		return this.http.post(this._apiUrl + "/invitation?link=" + token, {})
 			.pipe(
-				delay(2000),
 				src => this.requestWatcher.WatchFor(src)
-			)
-		}
-		return of(false)
-			.pipe(
-				delay(2000),
-				src => this.requestWatcher.WatchFor(src)
-			)
+			);
 	}
 
-	approveInvitation(id: number): Observable<boolean> {
-		let index = this.invitations.findIndex(i => i.id === id);
-
-		if (index > -1) {
-			let invitation = this.invitations[index];
-
-			this.users.push({spaceId: invitation.spaceId, userId: invitation.userId, isOwner: false, joinedAt: new Date(), role: invitation.role})
-
-			this.invitations.splice(index, 1);
+	changeRoleInInvitation(id: string, role: string): Observable<any> {
```

Comment:
Invitation id у нас number
```suggestion
	changeRoleInInvitation(id: number, role: string): Observable<any> {
```